### DISCUSSION TITLE
Remove ```IEnumerable``` Constraint for Collection Initializers

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#4089] [Language Ideas] [answered]
Not all classes that can ```Add``` should be ```IEnumerable```.  Today you have to create a dummy ```IEnumerable``` implementation in order to use collection initializers.  This requirement should be removed.

**Comment by YairHalberstadt** [comment]
Can you give an example of such a class

**Comment by ufcpp** [comment, marked as answer]
dup of https://github.com/dotnet/csharplang/discussions/455
